src/main/java/org/apache/ibatis/executor/resultset/DefaultResultSetHandler.java

```diff
@@ -710,10 +710,12 @@ private Optional<Constructor<?>> findConstructorForAutomapping(final Class<?> re
     if (constructors.length == 1) {
       return Optional.of(constructors[0]);
     }
-    for (final Constructor<?> constructor : constructors) {
-      if (constructor.isAnnotationPresent(AutomapConstructor.class)) {
-        return Optional.of(constructor);
-      }
+    List<Constructor<?>> automapConstructors = Arrays.stream(constructors).filter(x -> x.isAnnotationPresent(AutomapConstructor.class)).toList();
+    if (automapConstructors.size() > 1) {
+      throw new ExecutorException("@AutomapConstructor should be used in only one constructor.");
+    }
+    if (automapConstructors.size() == 1) {
+      return Optional.of(automapConstructors.get(0));
     }
     if (configuration.isArgNameBasedConstructorAutoMapping()) {
```

Comment:
How about using `reduce`?

```java
return Arrays.stream(constructors)
    .filter(x -> x.isAnnotationPresent(AutomapConstructor.class))
    .reduce((x, y) -> {
  throw new ExecutorException("@AutomapConstructor should be used in only one constructor.");
}).or(() -> {
  if (configuration.isArgNameBasedConstructorAutoMapping()) {
     ...
```

It probably is a matter of preference, so if you don't like it, the current version is fine. :)

Reply:
It looks very good! thank you.